src/webrtc/callEventHandler.ts

```diff
@@ -85,37 +88,16 @@ export class CallEventHandler {
         }
     }
 
-    private onEvent = (event: MatrixEvent) => {
-        // any call events or ones that might be once they're decrypted
+    private onEvent = async (event: MatrixEvent) => {
+        await this.client.decryptEventIfNeeded(event);
```

Comment:
So the call state machine relies on the call events arriving in the right order: I think think this will mean that the order they get processed in will be whatever order these promises resolve in, which *should* be the same order they were started in but I'm not really confident since it will depend on the js engine's microtask and indexeddb operation scheduling. Now we can just `await` for an event to be decrypted, we could probably just shove events into the buffer whether they're encrypted or not, then await on that promise serially for each event as we work our way through the buffer.